### DISCUSSION TITLE
GPU

### DISCUSSION THREAD

**Discussion by lnadolski** [atcollab/at#496] [Ideas]
Dear @lfarv, @swhite2401,

Do you have any medium or long term roadmaps for using pyat on PU clusters ?

What is your opinion regarding, performance, energy consumption?

Best regards,

Laurent.

**Comment by lfarv** [comment]
Hi @lnadolski,

Unfortunately I don't have any experience with GPUs, and no hardware for testing… I do not know which part of AT could benefit from that. In python, there is a "numpy-equivalent" package using CUDA, but I don't think AT spends most of its time in numpy. The real point in multi-particle tracking. It would be nice to have an expert look at that.

**Reply by lnadolski** [reply]
This will be indeed a strong asset for future.

**Comment by swhite2401** [comment]
Dear @lnadolski, I had a look at some point and found that openacc (https://www.openacc.org/) could potentially be implemented in a similar fashion as OpenMP in each passmethod.
This could be useful for DA and lifetime calculation, but I suspect it will not be for standard usage (optics, closed orbit) and I have no idea whether the overall gain would be worth the effort, this has to be carefully benchmarked.
